### PR TITLE
feat: SM-57 - create modal component

### PR DESCRIPTION
**Pull request by Suryansh-Simform**
# [Feature] Pull Request Modal Component

## Description

This pull request introduces a reusable `Modal` component to our application. The `Modal` component is designed to display content in a visually appealing and organized manner, with the ability to customize the header, content, and footer sections.

### Features

- **Customizable Header**: Allows setting a custom title or any other content for the modal's header.
- **Dynamic Content**: Supports any form of content, making the modal highly flexible and adaptable to different use cases.
- **Customizable Footer**: Provides options to include buttons or any other necessary controls in the modal's footer.
- **Cutomizable Modal close**: Provides options to close modal via outside click or not.
modal-dark
Modal-Light

### CODE REVIEW

**Review by kashyap27102 — CHANGES_REQUESTED** [review]

**Review by kashyap27102 — APPROVED** [review]

File: src/components/Modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  children: React.ReactNode;
+};
+
+export type ModalContentProps = ComponentProps<"div"> & {
+  children: React.ReactNode;
+};
+
+export type ModalFooterProps = ComponentProps<"div"> & {
+  children: React.ReactNode;
+};
```
Comment (adarshpandya-simform) [inline src/components/Modal/index.tsx:35]: The same type seems to be repeated for a couple of types. @Suryansh-Simform I wonder if we can have a single type in this case. If in the future, we see different use cases, we'll again separate them. Thoughts?

@kashyap27102 @anjali-sim @VatsalDave2102 @RiteshAdwani @svipulc ^^ If the following sounds good to you, please feel free to refactor your components as well 😄

File: src/components/Modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      {children}
+    </div>
+  );
+};
+
+Modal.Header = Header;
+Modal.Content = Content;
+Modal.Footer = Footer;
+
+export default Modal;
```
Comment (adarshpandya-simform) [inline src/components/Modal/index.tsx:125]: I'd personally prefer to have "named" exports

File: src/components/Modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+const Footer: React.FC<ModalFooterProps> = ({ children }) => {
+  return <div className="p-4">{children}</div>;
+};
+
+Modal.Body = Body;
+Modal.Header = Header;
+Modal.Content = Content;
+Modal.Footer = Footer;
+
+export default Modal;
```
Comment (VatsalDave2102) [inline src/components/Modal/index.tsx:125]: I think we have been using named exports for other components as well.

Comment (adarshpandya-simform) [reply src/components/Modal/index.tsx:125]: @VatsalDave2102 https://github.com/svipulc/Evoke-ui/pull/10#discussion_r1740316101 Us 🤝

**Pull request merged**